### PR TITLE
 - Adding docs for Calvin's profiling technique

### PR DESCRIPTION
**Pull request by ilooner**
### Context

Adding documentation for profiling mantis workers.

### Checklist

- [x] `./gradlew build` compiles code correctly
- [x] Added new tests where applicable
- [x] `./gradlew test` passes all tests
- [x] Extended README or added javadocs where applicable
- [x] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ nav:
     - 'Tuning': 'operate/tuning.md'
     - 'Autoscaling': 'operate/autoscaling.md'
     - 'Autoscaling Strategies': 'operate/autoscalingstrategies.md'
+    - 'Memory and CPU Profiling': 'operator/profiling.md'
```
Comment (jeffchao) [inline docs/mkdocs.yml:33]: `operator` -> `operate`

File: docs/docs/operate/profiling.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+can be helpful for debugging OOMs. Below are the steps to start profiling your running Mantis job:
+
+ 1. Download a profiling tool. We recommend [Zulu Mission Control (ZMC)](https://www.azul.com/products/zulu-mission-control/).
+ 1. Untar ZMC
+    ```
+    tar -xzvf <zulu_mission_control_package>.tar.gz
+    ```
+ 1. Run zmc.
+    ```
+    note
```
Comment (jeffchao) [inline docs/docs/operate/profiling.md:11]: Does this work? I think it needs to be `!!! note`

File: docs/docs/operate/profiling.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    <img src="../images/profiling/workerDebugURL.png" alt="Worker Debug URL" width="600"/>
+
+ 1. Setup an SSH tunnel with local portforwarding for your worker and port. The tunnel command for the example worker
+    in the previous step would be:
+    ```
+    ssh -L 7156:100.82.159.148:7156 100.82.159.148
+    ```
+ 1. After setting up local port forwarding. Add the worker JVM to ZMC by clicking the plug and star icon in the top right corner.
+
+    <img src="../images/profiling/zmcAddVM.png" alt="ZMC add VM" width="600"/>
```
Comment (jeffchao) [inline docs/docs/operate/profiling.md:30]: Images can be done in markdown. Example: `Job running`

File: docs/docs/operate/profiling.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    ```
+    ```
+    <your dir>/<zulu_mission_control_package>/Zulu\ Mission\ Control.app/Contents/MacOS/zmc
+    ```
+ 1. Find a worker you want to debug in your job and click on it to get its **ip address** and **debug port** In this
+    example these are `100.82.159.148` and `7156` respectively.
+
+    <img src="../images/profiling/workerDebugURL.png" alt="Worker Debug URL" width="600"/>
+
+ 1. Setup an SSH tunnel with local portforwarding for your worker and port. The tunnel command for the example worker
```
Comment (calvin681) [inline docs/docs/operate/profiling.md:23]: should the tunnel command be `ssh -L 7156:127.0.0.1:7156 100.82.159.148` for accessing the port locally? Also, may be a note that this is only needed if the worker ip cannot be hit directly.

**Comment by calvin681** [comment]
This is great, thanks for adding this.

File: docs/docs/operate/profiling.md (hunk 4 above)
Comment (jeffchao) [reply docs/docs/operate/profiling.md:23]: This. I just do this normally. https://gist.github.com/jeffchao/ac392c016aff17977c890f4c3101c7f2

**Review by jeffchao — CHANGES_REQUESTED** [review]
Left some comments.

File: docs/docs/operate/profiling.md (hunk 4 above)
Comment (ilooner) [reply docs/docs/operate/profiling.md:23]: Added note about direct access. Switched to using 127.0.0.1 . I think that's the destination ip. So if the worker's debug port is listening on 127.0.0.1 or it's host ip, then both should work as long as your sshing into the same host. Tested with our setup and both worked.

File: docs/docs/operate/profiling.md (hunk 3 above)
Comment (ilooner) [reply docs/docs/operate/profiling.md:30]: I noticed that images get stretched weirdly with the markdown image reference in other mkdocs projects, so I've used `` to control the width in the past. Will switch to using the markdown references, hopefully it'll come out okay.

File: docs/docs/operate/profiling.md (hunk 2 above)
Comment (ilooner) [reply docs/docs/operate/profiling.md:11]: Thanks for catching this, fixing

File: docs/mkdocs.yml (hunk 1 above)
Comment (ilooner) [reply docs/mkdocs.yml:33]: Doh, thanks for catching.

**Review by jeffchao — APPROVED** [review]
LGTM.

**Pull request merged**